code/modules/mob/living/silicon/silicon_mob.dm

```diff
@@ -439,3 +474,131 @@
+		hat_icon_state = silicon_hat.icon_state
+	if(isnull(hat_alpha))
+		hat_alpha = silicon_hat.alpha
+	if(!hat_color)
+		hat_color = silicon_hat.color
+
+	head_icon = get_hat_overlay()
+
+	add_overlay(head_icon)
+
+/mob/living/silicon/proc/get_hat_overlay()
+	if(!(hat_icon_file || hat_icon_state))
+		return
+	var/image/borgI = image(hat_icon_file, hat_icon_state)
+	borgI.alpha = hat_alpha
+	borgI.color = hat_color
+	borgI.pixel_y = hat_offset_y
+	if(!is_centered)
+		borgI.transform = matrix(1.125, 0, 0.5, 0, 1, 0)
+	return borgI
+
+/**
+  * Attempts to put an item on a silicon's head.
+  *
+  * Arguments:
+  * * item_to_add - The item we're attempting to place on a silicon.
+  * * user - Mob trying to put a hat on a silicon.
+  * Returns boolean reflecting if a hat was succesfully placed on the silicon.
+  */
+/mob/living/silicon/proc/place_on_head(obj/item/item_to_add, mob/user)
+	if(!item_to_add)
+		user.visible_message(
+			"<span class='notice'>[user] pats [src] on the head.</span>",
+			"<span class='notice'>You pat [src] on the head.</span>")
+		if(flags_2 & HOLOGRAM_2)
+			return FALSE
+		return FALSE
+
+	if(!istype(item_to_add, /obj/item/clothing/head))
+		to_chat(user, "<span class='warning'>[item_to_add] cannot be worn on the head by [src]!</span>")
+		return FALSE
+
+	if(!can_be_hatted)
+		to_chat(user, "<span class='notice'>No matter how hard you try you don't seem to be able to put a hat on [src]!</span>")
+		return FALSE
+
+	if(silicon_hat)
+		to_chat(user, "<span class='warning'>[src] can't wear more than one hat!</span>")
+		return FALSE
+
+	if(!can_wear_restricted_hats && is_type_in_list(item_to_add, restricted_hats))
+		to_chat(user, "<span class='warning'>[item_to_add] does not fit on the head of [src]!</span>")
+		return FALSE
+
+	if(user && !user.unEquip(item_to_add))
+		to_chat(user, "<span class='warning'>[item_to_add] is stuck to your hand, you cannot put it on [src]!</span>")
+		return FALSE
+
+	user.visible_message(
+		"<span class='notice'>[user] puts [item_to_add] on [real_name].</span>",
+		"<span class='notice'>You put [item_to_add] on [real_name].</span>"
+	)
+	item_to_add.forceMove(src)
+	silicon_hat = item_to_add
+	update_icons()
+
+	return TRUE
+
+/**
+  * Attempts to remove any hats a silicon is wearing.
+  *
+  * Arguments:
+  * * user - Mob trying to remove a silicon's hat.
+  * Returns boolean reflecting if a hat was successfully removed from the silicon.
+  */
+/mob/living/silicon/proc/remove_from_head(mob/user)
+	if(silicon_hat)
+		if(silicon_hat.flags & NODROP)
+			to_chat(user, "<span class='warning'>[silicon_hat.name] is stuck on [src]'s head, it is impossible to remove!</span>")
+			return TRUE
+
+		to_chat(user, "<span class='warning'>You remove [silicon_hat.name] from [src]'s head.</span>")
+		user.put_in_hands(silicon_hat)
+
+		null_hat()
+
+		update_icons()
+	else
+		to_chat(user, "<span class='warning'>[src] isn't wearing anything on their head!</span>")
+		return FALSE
+
+	return TRUE
```

Comment:
```suggestion
	if(!silicon_hat)
		to_chat(user, "<span class='warning'>[src] isn't wearing anything on their head!</span>")
		return FALSE
	if(silicon_hat.flags & NODROP)
		to_chat(user, "<span class='warning'>[silicon_hat.name] is stuck on [src]'s head, it is impossible to remove!</span>")
		return TRUE

	to_chat(user, "<span class='warning'>You remove [silicon_hat.name] from [src]'s head.</span>")
	user.put_in_hands(silicon_hat)

	null_hat()
	update_icons()

	return TRUE
```
Think it might be more readable like that?